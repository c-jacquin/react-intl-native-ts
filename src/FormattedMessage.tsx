import React, { SFC } from 'react'
import { FormattedMessage } from 'react-intl'
import { Text } from 'react-native'

import { FormattedMessageProps } from './types'

const NativeFormattedMessage: SFC<FormattedMessageProps> = props => (
  <FormattedMessage {...props}>{(localized: string) => <Text style={props.style}>{localized}</Text>}</FormattedMessage>
)

export default NativeFormattedMessage
