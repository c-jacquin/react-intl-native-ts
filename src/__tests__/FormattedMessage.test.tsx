import * as React from 'react'
import { Text } from 'react-native'
import * as renderer from 'react-test-renderer'
import { IntlProvider } from 'react-intl'

import FormattedMessage from '../FormattedMessage'

describe('FormattedMessage', () => {
    it('should renders correctly', () => {
        const tree = renderer
            .create(
                <IntlProvider locale="en" messages={{  bar: 'foo' }}>
                    <FormattedMessage style={{}} id={'bar'} />
                </IntlProvider>
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})
