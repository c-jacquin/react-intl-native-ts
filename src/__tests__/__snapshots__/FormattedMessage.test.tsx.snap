// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormattedMessage should renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={Object {}}
>
  foo
</Text>
`;
